### DISCUSSION TITLE
Companion app as media player

### DISCUSSION THREAD

**Discussion by White-Goblin** [music-assistant/support#396] [Feature requests and ideas]
I would love if the companion apps would create a media player entity so we could use this nice new music manager to stream music while in the car, headphones in bed, etc. If I had this I could seriously consider retiring my jellyfin server.

Bonus points if it supports local song caching for offline listening, but that might be being the scope of this project.

**Comment by marcelveldt** [comment]
This project once started as a standalone project. The ability to have a local player within the webapp was included once and in fact I didn't port it when the project moved to being a part of HA.

So long story short I think this is perfectly doable!

**Comment by jazzyisj** [comment]
Just an FYI if you haven't tried it, [browser mod media players](https://github.com/thomasloven/hass-browser_mod) work just dandy with Music Assistant.  Tested on mobile in the companion app for android.

**Reply by White-Goblin** [reply]
Thanks for the lead.

I've tested it out and browser mod seems way too finicky for me. I've now got 6 hardware/browser devices, 4 of which are duplicates. I can't delete any of these new devices or their entities when I try calling browser mod's service because I get an error. Since this is a yaml managed component, I can't manage the devices from the UI either. After finally getting my phone's device/browser combo to work (caching issues with the companion app), there's no sound even though home assistant says the media is playing.

Browser mod seems to be lean heavily towards the tinker side of the usability<---->endless tinkerer hell spectrum, and I try to stay away from that stuff.

**Reply by marcelveldt** [reply]
Browser mod is not a solution but more of a temporary workaround imo.
It is on my wish list too bring a built-in player to the MA frontend. Other priorities now but I'm pretty sure this will come some day :-)

**Reply by WebfinityDesign** [reply]
A built-in player to the MA frontend would be nice. That's something I'm missing now when I want to play MA on my desktop PC when I'm working on it.

**Reply by OzGav** [reply]
Look at Marcel’s reply below about downloading squeezelight. You can also search for squeeze light compatible players

**Reply by WebfinityDesign** [reply]
I just did. It works! Good solution

**Comment by Jdwoodson** [comment]
Big vote for this. Trying to find the best way to continue using MA on my office PC that has nice speaker setup instead of the dinky Google Home speaker on the desk! I think this would be the easiest solution...

**Reply by marcelveldt** [reply]
Temporary solution: [download squeezelite](https://sourceforge.net/projects/lmsclients/files/squeezelite/) and run it.
Activate the "slimproto" integration in HA and it will auto detect the squeezelite player.
Enjoy

**Reply by Jdwoodson** [reply]
you're a rockstar - thank you so much for this idea!

**Comment by Greylinux** [comment]
This would be awesome if it was integrated into music assistant 2.0

**Comment by Jessidy62** [comment]
Hey guys, I'm really happy with Music assistant running along with Home Assistant on my Raspberry Pi.
Congrats for the huge work: stability, interface look, features with multiple music sources, etc...
I did try several solution to play my music ( mainly coming from my NAS and more recently from my Spotify account) on my iPhone or iPad, listening it on my Bose QC35. I know, I know, final bluetooth encoding is not the best solution for sound quality, but this is very easy and confortable for me, and others around me ;-)
After many attempts to use Browser Mod players, or squeezelite + slimproto, that I did not find stables, I ended up with using Snapcast and snapweb clients. But the user experience is not yet at its best, since we need to keep a web page with snapweb open.
Also, except if I did not understand something (this is perfectly possible :) ), it does not really work with HA companion app. We need to be connected to Home assistant through a web browser, to keep that snapweb "alive".
to make it short, Music Assistant definitely deserve an embedded media player, compatible with the HA companion App.
@marcelveldt , you did talk about it almost 2 years ago ; could we have an update about that potential new feature, please?

Thanks a lot!

**Reply by OzGav** [reply]
There is no update as dev resources remain scarce. The squeezelite players are working fine for just about everyone based on the lack of issue reports?

**Reply by White-Goblin** [reply]
I use lms running in docker with an iframe to get it into home assistant. That’s for whole home audio.  For mobile playback I use jellyfin and a mobile app such as Finamp for offline caching. Unfortunately I couldn’t make music companion work so I’m using lms instead.

**Comment by OzGav** [comment]
The first iteration of this is in 2.5.0b11

**Reply by Jessidy62** [reply]
Thank you OzGav! I’m eager to test it, as soon as I can 😊

**Comment by AkoreJepade** [comment]
What is the current status on this now that sendspin is being incorporated into MA? I have some Lenovo tablets and a Lenovo SmartThink that have HA running on them but they are only able to play music while the MA is open. As soon as I go to another dashboard, even if that dashboard has MA controls on it, the music cuts off and the play no longer shows in MA. IS there a setting I am overlooking to keep MA services always running on the devices even if I leave MA/HA or while I am in HA but on another dashboard?

**Reply by White-Goblin** [reply]
I think you’re looking for a sendspin client. I know there’s one in beta that can run as a service:

https://github.com/Sendspin/sendspin-cli

**Reply by AkoreJepade** [reply]
That seems to be more for Linux. There is a [https://github.com/chrisuthe/SendspinDroid](url) for Android that I have been playing with but I would really like to see something built right into MA/HA that keeps the Sendspin service running in the background so that you do not have 100's of Sendspin instances showing up in MA. But since this is all in Alpha stages at the moment I am sure not far down the road it will all come together in a great working version... in fact I heard a big update to MA should be released in the next week or 2 so fingers crossed. I am moving away from Sonos so I have very high hopes for the futures of MA/Sendspin as a replacement.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
